index.bs

```diff
@@ -707,6 +711,37 @@ NOTE: The <a href="https://immersive-web.github.io/layers">WebXR layers module</
 
 </div>
 
+<div class="algorithm" data-algorithm="apply-target-frame-rate">
+
+When requested, the {{XRSession}} |session| MUST <dfn>apply the target frame rate</dfn> |rate| and an optional |promise| by running the following steps:
+  1. [=Queue a task=] to perform the following steps:
+    1. Pass the new target frame rate |rate| to the [=XR Compositor=].
+    1. The [=XR Compositor=] MAY use |rate| to update the {{XRSystem}}'s frame rate.
+    1. Set the |session|'s internal [=XRSession/framerate=] to the {{XRSystem}}'s recommended frame rate. It is up to the {{XRSystem}} to determine this recommended frame rate.
+    1. [=Queue a task=] to perform the following steps:
+        1. Await until the {{XRSystem}}'s frame rate has taken effect.
```

Comment:
We should perhaps be more precise here that we are waiting until the XR compositor makes all changes it intends to make at this time in response to the request.  This accommodates the case where the device is already at 30fps and stays there - no change was made, but the XR compositor is ready to respect that request when the temporary condition is lifted.

```suggestion
        1. If the [=XRSession/framerate=] was changed, await until the {{XRSystem}}'s frame rate has taken effect.
```